GitUI/CommandsDialogs/BrowseDialog/FormUpdates.cs

```diff
@@ -145,11 +148,27 @@ private void linkChangeLog_LinkClicked(object sender, LinkLabelLinkClickedEventA
             Process.Start("https://github.com/gitextensions/gitextensions/blob/master/GitUI/Resources/ChangeLog.md");
         }
 
-        private void btnDownloadNow_Click(object sender, EventArgs e)
+        private async void btnDownloadNow_Click(object sender, EventArgs e)
         {
             try
             {
-                Process.Start(UpdateUrl);
+                progressBar1.Visible = true;
+                btnDownloadNow.Enabled = false;
+                UpdateLabel.Text = _downloadingUpdate.Text;
+
+                WebClient webClient = new WebClient();
```

Comment:
```suggestion
                using (var webClient = new WebClient()) {
```

Reply:
Good point, although we close the app if all is good. If it fails, then yes GC would need to cleanup in this case. I'll make the changes :)

Reply:
Done